nixos/tests/postgresql.nix

```diff
@@ -1,26 +1,44 @@
-import ./make-test.nix ({ pkgs, ...} : {
+import ./make-test.nix ({ pkgs, ...} : 
+let
+  postgresql-versions = pkgs.callPackages ../../pkgs/servers/sql/postgresql { };
+  test-sql = pkgs.writeText "postgresql-test" ''
+    CREATE EXTENSION pgcrypto; -- just to check if lib loading works
+    CREATE TABLE sth (
+      id int
+    );
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+    INSERT INTO sth (id) VALUES (1);
+  '';
+in
+{
   name = "postgresql";
   meta = with pkgs.stdenv.lib.maintainers; {
     maintainers = [ zagy ];
   };
 
-  nodes = {
-    master =
-      { pkgs, config, ... }:
-
+  nodes = pkgs.lib.mapAttrs' (k: v: {name=k; value=
+      {pkgs, config, ...}:
       {
+        services.postgresql.package=v;
         services.postgresql.enable = true;
-        services.postgresql.initialScript =  pkgs.writeText "postgresql-init.sql"
-          ''
-          CREATE ROLE postgres WITH superuser login createdb;
-          '';
-      };
-  };
+      };}) 
+      postgresql-versions;
 
-  testScript = ''
-    startAll;
-    $master->waitForUnit("postgresql");
-    $master->sleep(10); # Hopefully this is long enough!!
-    $master->succeed("echo 'select 1' | sudo -u postgres psql");
-  '';
+  testScript = builtins.concatStringsSep "\n" (map (p: ''
+    ${"$"}${p}->start;
```

Comment:
@basvandijk for sure I tried that! but didn't work, there is some overlap with perl but didn't investigated for too long :-)
I did again:
```
  testScript = builtins.concatStringsSep "\n" (map (p: ''
    $${p}->start;
    ${"$"}${p}->waitForUnit("postgresql");
```
and this renders to
```
$${p}->start;
$postgresql96->waitForUnit("postgresql");
```
and throws and error:
```
error: Global symbol "$p" requires explicit package name (did you forget to declare "my $p"?) at (eval 15) line 1.
```